eel/src/event_handler.rs

```diff
@@ -203,7 +203,20 @@ impl EventHandler for LipaEventHandler {
                     .unwrap();
             }
             Event::PaymentPathSuccessful { .. } => {}
-            Event::PaymentPathFailed { .. } => {}
+            Event::PaymentPathFailed {
+                payment_id,
+                payment_hash: _,
+                payment_failed_permanently: _,
+                network_update: _,
+                all_paths_failed,
+                ..
+            } => {
+                if all_paths_failed {
+                    if let Some(payment_id) = payment_id {
+                        self.channel_manager.abandon_payment(payment_id);
+                    }
+                }
+            }
```

Comment:
This may not be what we want to do. The `InvoicePayer` (see [docs](https://docs.rs/lightning-invoice/0.21.0/lightning_invoice/payment/index.html)) should be capable of reacting to failed payment paths and retrying a few times before failing. 
What is being done here may conflict with its logic. Maybe we have a problem with the `InvoicePayer` setup?